tempesta_fw/t/unit/sse_parser/sse_parser.c

```diff
@@ -0,0 +1,1552 @@
+    i->position = buf;
+    i->readlen = size;
+}
+
+inline int inputIteratorReadable(InputIterator * restrict i)
+{
+    //test if there are >=16 bytes or there is a newline
+    int q = 1;
+    if ((i->bytesin + i->readlen) < 16) {
+        __m128i r = _mm_rm(32 - i->bytesin);
+        __m128i b = _mm_bm(32 - i->bytesin);
+
+        Vector v = _mm_blend(_mm_shuffle_epi8(i->latch[0], r),
+                         _mm_shuffle_epi8(i->latch[1], r),
+                         b);
+        v = _mm_cmpeq_epi8(v, cc._n);
+        q = _mm_movemask_epi8(v);
+    }
+    return q;
+}
+
+inline Vector readIterator(InputIterator * restrict i)
+{
+    if (i->bytesin < 16) {
+        if (unlikely(i->readlen < 16)) {
+            char tmp[16];
+            int j;
+            for(j = 0; j < i->readlen; ++j)
+                tmp[j] = i->position[j];
+            i->readlen = 0;
+            i->position += j;
+
+            __m128i r = _mm_rm(j);
+            __m128i b = _mm_bm(j);
+            __m128i v1 = _mm_shuffle_epi8(_mm_lddqu_si128((__m128i*)tmp), r);
+            __m128i v2 = _mm_shuffle_epi8(i->latch[1], r);
+            i->latch[1] = _mm_blend(v2, v1, b);
+            i->latch[0] = v2;
+            i->bytesin += j;
+        } else {
+            i->latch[0] = i->latch[1];
+            i->latch[1] = _mm_lddqu_si128((const __m128i*)i->position);
+            i->bytesin += 16;
+            i->readlen -= 16;
+            i->position += 16;
+        }
+    }
+    __m128i r = _mm_rm(32 - i->bytesin);
+    __m128i b = _mm_bm(32 - i->bytesin);
+
+    return _mm_blend(_mm_shuffle_epi8(i->latch[0], r),
+                     _mm_shuffle_epi8(i->latch[1], r),
+                     b);
+}
+
+inline int  containsNewline(Vector v) {
+    __m128i n = _mm_cmpeq_epi8(v, cc._n);
+    return _mm_movemask_epi8(n);
+}
+
+inline SymbolMap createEmptySymbolMap() {
+   return _mm_setzero_si128();
+}
+
+SymbolMap createSymbolMapFromCharset(const char * cs) {
+    char set[16];
+    int i;
+    for(i = 0; i < 16; ++i)set[i] = 0;
+    while(*cs) {
+        if (cs[0] & 0x80) continue;
+        set[cs[0] & 0x0F] |= (1<<(cs[0]>>4));
+        ++cs;
+    }
+    return _mm_lddqu_si128((__m128i*)set);
+}
+
+SymbolMap appendSymbolMapFromCharset(SymbolMap sm, const char * cs) {
+    return _mm_or_si128(sm, createSymbolMapFromCharset(cs));
+}
+
+inline Vector matchSymbolsMask(SymbolMap sm, Vector v) {
+    __m128i mask1 = _mm_shuffle_epi8(sm, v);
+    __m128i mask2 = _mm_and_si128(
+                cc.masktest2,
+                _mm_srli_epi16(v, 4));
+    __m128i mask3 = _mm_shuffle_epi8(
+                cc.masktest1,
+                mask2);
+    __m128i mask4 = _mm_and_si128(
+                mask3,
+                mask1);
+    __m128i vec1 = _mm_cmpgt_epi8(
+                mask4,
+                _mm_setzero_si128());
+    __m128i vec2 = _mm_cmplt_epi8(
+                mask4,
+                _mm_setzero_si128());
```

Comment:
unnecessary line breaks between function arguments are hard to read and doesn't comply our Coding Style.